sites/kit.svelte.dev/src/routes/home/Video.svelte

```diff
@@ -0,0 +1,128 @@
+<script>
+	import testimonials from './testimonials.mp4';
+	import volume_off from './volume-off.svg';
+	import volume_high from './volume-high.svg';
+	import { onMount } from 'svelte';
+
+	/** @type {HTMLVideoElement} */
+	let video;
+	let muted = true;
+	let unused = true;
+
+	let d = 0;
+	let t = 0;
+
+	onMount(() => {
+		// I think this binding is broken in SSR because the video already
+		// has a duration by the time hydration occurs. TODO investigate
+		d = video.duration;
+
+		const observer = new IntersectionObserver(
+			(entries) => {
+				for (const entry of entries) {
+					if (entry.isIntersecting) {
+						video.play();
+						observer.disconnect();
+					}
+				}
+			},
+			{
+				threshold: 1
+			}
+		);
+
+		observer.observe(video);
+
+		return () => {
+			observer.disconnect();
+		};
+	});
+</script>
+
+<div class="video-player">
+	<video
+		src={testimonials}
+		loop
+		bind:this={video}
+		bind:muted
+		bind:currentTime={t}
+		bind:duration={d}
+		on:click={() => {
+			if (video.paused) {
+				video.play();
+			} else {
+				video.pause();
+			}
+		}}
```

Comment:
We should make sure the video is pausable/playable via keyboard as well.

Reply:
Videos aren't focusable by default, so should we listen for spacebar presses on `window`? Presumably we don't want to `preventDefault()`, but we also don't want the spacebar to toggle the video _and_ scroll.

Or do we make the video focusable?

Reply:
I actually think a visible play/pause button is the best solution, since then it's very clear how to play/pause the video 👍